index.html

```diff
@@ -162,12 +180,89 @@ <h3>
         The <a>type</a> member of an <a>ImageResource</a> is a <a>string</a> that is a <a>valid MIME type</a>. It acts
         as a hint to the media type of the image. The purpose of this member is to allow the user agent to ignore
         images of media types it does not support.
-
+      </p>
+      <p>
         There is no default <a>MIME type</a> for an <a>ImageResource</a>. However, for the purposes of <a>determining
           the type of the resource</a>, user agents must expect the resource to be an image.
       </p>
+      <p>
+          The steps for <dfn>processing the <code>type</code> member of an
+          image</dfn> are given by the following algorithm. The algorithm takes
+          an <var>image</var> object as an argument, and returns either a
+          <a>string</a> or <code>undefined</code>.
+        </p>
+        <ol>
+          <li>Let <var>value</var> be <var>image</var>["<a>type</a>"].
+          </li>
+          <li>If <a>Type</a>(<var>value</var>) it not String, return
+          <code>undefined</code>.
+          </li>
+          <li>If <var>value</var> is not a <a>valid MIME type</a> or the value
+          of <var>type</var> is not a supported media format, issue a developer
+          warning and return <code>undefined</code>.
+          </li>
+          <li>Return <var>value</var>.
+          </li>
+        </ol>
     </section>
   </section>
+  <section>
+    <h3>
+      Processing an array of image resources
+    </h3>
+    <p>
+      The steps for <dfn>processing <a>ImageResource</a> members</dfn> are
+      given by the following algorithm. The algorithm takes a
+      <a>URL</a> <var>base URL</var>, which is the URL from which the
```

Comment:
"which is the URL from which the ImageResource was fetched" isn't really necessary or correct.

For one thing, you don't fetch an image resource from a URL, it's found inside something else such as a manifest or perhaps just passed as an object to a JavaScript API. Really, this needs to say it's the URL of whatever context the invoking API is being called from, but that's very hand-wavey. We don't need formal text for this, it's just an argument which the caller can set to whatever it wants.

Suggest taking out this sentence and maybe adding a note box saying "base URL typically represents the URL of the document that triggered the processing of the ImageResource, for example the document in which a method was called, or the `<a>`manifest`</a>` containing it." Something like that.

(Edit: Ah I see, this was copied from Manifest. I just deleted this sentence in the Manifest version in w3c/manifest#811)

Reply:
Looking over this this morning, I had an idea. Let me know what you think: instead of being hand-wavey & such, we could just call it a URL "`context`". Then, as you mention, we could include a note defining that as the base URL of a document containing it or the execution scope of the method making use of this API.

Reply:
I’ve merged in your relevant changes.